Add support for XP-Pen Artist 12 3rd Generation

quotejericho opened the pull request:
Hello!

This is a draft PR to help add support for the XP-Pen Artist 12 3rd Gen, and the format is based on the Artist 12 2nd Gen's configuration.

Some missing information:

- [ ] **Digitizer**
     - [ ]  Width (Official Specs Added)
     - [ ]  Height (Official Specs Added)
     - [ ] MaxX
     - [ ] MaxY
- [x] **Pen**
     - [x] MaxPressure (Official Added)
     - [X] ButtonCount (Official Added)
- [x] **Wheel 1**
     - [x] Step Count
     - [x] IsRelative
     - [x] Angle of Zero Reading
     - [X] Buttons
- [ ] **Wheel 2**
     - [ ] Step Count
     - [ ] IsRelative
     - [ ] Angle of Zero Reading
     - [X] Buttons
- [X] **AuxiliaryButtons**
     - [X] ButtonCount (Official Added)
- [ ] **DigitizerIdentifiers**
     - [X] VendorID (Added via lsusb)
     - [X] ProductID (Added via lsusb)
     - [x] InputReportLength
     - [x] OutputReportLength
     - [x] ReportParser
     - [x] OutputInitReport
- [x] Attributes
     - [x] libinputoverride (unsure if correct)
     
 I am unsure of the width, height, nor maxPressure being correct as the values for the 2nd generation were slightly off from their official values. Unfortunately despite having the tablet I am struggling to get anything to be read from it using OpenTabletDriver. I know there is a guide on the wiki but I don't know where I am supposed to get access to the parser and such. Apologies for the amateur PR but I hope to be able to help add support for this tablet in any way.

quotejericho committed:
Add configuration for XP-Pen Artist 12 (3rd Gen)

quotejericho committed:
Add support for XP-Pen Artist 12 (3rd Gen)

gonX commented on OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 12 (3rd Gen).json:
```diff
@@ -0,0 +1,31 @@
+      "MaxY": 29616
+    },
+    "Pen": {
+      "MaxPressure": 16384,
```
```suggestion
      "MaxPressure": 16383,
```
this is usually 1 below the pressure range stated by the manufacturer, as `0` is also a valid pressure number

`grep` shows:
```
$ grep -R 16383
Gaomon/M8 (Variant 2).json:      "MaxPressure": 16383,
Huion/HC16 (Variant 2).json:      "MaxPressure": 16383,
Huion/Kamvas Pro 27.json:      "MaxPressure": 16383,
Huion/L310.json:      "MaxPressure": 16383,
Huion/Kamvas 13 (Gen 3).json:      "MaxPressure": 16383,
Huion/Kamvas 16 (Gen 3).json:      "MaxPressure": 16383,
Huion/Kamvas Pro 19 (4K).json:      "MaxPressure": 16383,
Huion/L610.json:      "MaxPressure": 16383,
XP-Pen/Deco Pro LW Gen2.json:      "MaxPressure": 16383,
XP-Pen/Deco Pro XLW Gen2.json:      "MaxPressure": 16383,
XP-Pen/Deco 640 (IT640).json:      "MaxPressure": 16383,
XP-Pen/Deco Pro MW Gen2.json:      "MaxPressure": 16383,
<snip>
```

gonX reviewed (commented):
Nothing out of the ordinary but would of course like this to be verified first

quotejericho committed:
Update Artist 12 (3rd Gen).json

Co-authored-by: Sebastian Jensen <gonx@gonx.dk>

quotejericho committed:
OTD: Corrected Input/Output Report Length + Wheel Draft

quotejericho committed:
OTD: XP-Pen updated configuration (crono141)

quotejericho committed:
Change support status for XP-Pen Artist 12 (3rd Gen)

Updated support status for XP-Pen Artist 12 (3rd Gen) to indicate missing features.
